### ISSUE TITLE
Reduce/Eliminate User Prompts for Keychain Access

### ISSUE THREAD

**Issue by lyoung-confluent** [common-fate/granted#403]
When using `granted`/`assume` users may experience prompts for their password to access the Keychain items, particularly on macOS:
<img width="546" alt="image" src="https://user-images.githubusercontent.com/91491244/231655585-0a9b82ed-b423-4f86-abd6-76e1e84a0ad1.png">
This can be reproduced easily on macOS by running `assume` for a valid AWS SSO profile then running `granted sso-tokens clear --all` to clear out the entry.

I suspect this is related to some combination in an oddity with the code signing of the 2 separate binaries, the ACLs on the keychain items or the significantly older version of https://github.com/99designs/go-keychain used by `granted`, however more investigation is required.

**Comment by lyoung-confluent** [comment]
Upon further investigation using https://github.com/its-a-feature/LockSmith (see also [Lock Picking the macOS Keychain talk at OBTS 5.0](https://www.youtube.com/watch?v=jKE1ZW33JpY)), we can see the ACLs on a keychain item created using the signed `v0.10.0` release:
```
Account:       https://<snip>.awsapps.com/start
Label:         granted-aws-sso-tokens
Service:       granted-aws-sso-tokens
Creation Date: 2023-04-12 21:08:51
Modify Date:   2023-04-12 21:08:51
Class:         genp
Owner Authorizations based on ACLAuthorizationPartitionID
Owner Authorizations:
    ACLAuthorizationEncrypt
    ACLAuthorizationDecrypt
    ACLAuthorizationDerive
    ACLAuthorizationExportClear
    ACLAuthorizationExportWrapped
    ACLAuthorizationMAC
    ACLAuthorizationSign
    ACLAuthorizationIntegrity
    ACLAuthorizationPartitionID
------------ACLS------------
    --- Entry 0
        Description: (null)
        All applications are trusted
        Authorizations:
             ACLAuthorizationEncrypt
    --- Entry 1
        Description:
        Trusted App: /usr/local/Cellar/granted/0.10.0/bin/assumego
            Requirement String: identifier assumego and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = VXC4RPMSUJ
        Authorizations:
             ACLAuthorizationDecrypt
             ACLAuthorizationDerive
             ACLAuthorizationExportClear
             ACLAuthorizationExportWrapped
             ACLAuthorizationMAC
             ACLAuthorizationSign
    --- Entry 2
        Description: 80912843e85942c565983c435cfa730ad00ce0664d7671d7da763950f0848f5f
        All applications are trusted
        Authorizations:
             ACLAuthorizationIntegrity
    --- Entry 3
        Allowed Code Signatures: teamid:VXC4RPMSUJ
            TeamID doesn't match current application: nil
        Description: 3c3f786d6c2076657273696f6e3d22312e302220656e636f64696e673d225554462d38223f3e0a3c21444f435459504520706c697374205055424c494320222d2f2f4170706c652f2f44544420504c49535420312e302f2f454e222022687474703a2f2f7777772e6170706c652e636f6d2f445444732f50726f70657274794c6973742d312e302e647464223e0a3c706c6973742076657273696f6e3d22312e30223e0a3c646963743e0a093c6b65793e506172746974696f6e733c2f6b65793e0a093c61727261793e0a09093c737472696e673e7465616d69643a5658433452504d53554a3c2f737472696e673e0a093c2f61727261793e0a3c2f646963743e0a3c2f706c6973743e0a
        All applications are trusted
        Authorizations:
             ACLAuthorizationPartitionID
    --- Entry 4
        Description:
        No applications are trusted
        Authorizations:
             ACLAuthorizationChangeACL
```
As expected, this is the default ACL that macOS populates on new keychain items if the application does not otherwise specify.

**Comment by lyoung-confluent** [comment]
The root-cause appears to be related to the ACL on the keychain item:
> Requirement String: **identifier assumego** and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = VXC4RPMSUJ

macOS is setting the ACL so only binaries named `assumego` signed by the Common Fate code signing certificate can access this item. Because `granted` is a _different_ binary (although signed by the same certificate) it fails this check triggering a prompt for the user to grant access.

**Comment by lyoung-confluent** [comment]
One solution for this would be to use an ["access group"](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps) to grant multiple binaries signed by the same certificate access to the keychain items. This does not appear to be easily configurable via the keyring library and could introduce complexity for migrating existing keychain items. 

A simpler solution that would also be cross-platform would be to use a single binary for both `granted`/`assumego` and automatically change the behavior based on the filename provided as `os.Args[0]`. This would probably mean `granted` is the actual binary and `assume` is a symlink to the `granted` binary. This may introduce some complexity into the installer/packages such as [the homebrew formula](https://github.com/common-fate/homebrew-granted/blob/main/Formula/granted.rb) but could also cutdown on the overall package/release size.

**Comment by lyoung-confluent** [comment]
This was resolved via #404 and released as [v0.11.0](https://github.com/common-fate/granted/releases/tag/v0.11.0)
